File: JAVA/program/Human.java


public class Human {
    // Attribut protected, untuk diwariskan
    protected String nik;
    protected String nama;
    protected String jenis_kelamin;

    // Parameterized constructor
    public Human(String nik, String nama, String jenis_kelamin) {
        this.nik = nik;
        this.nama = nama;
        this.jenis_kelamin = jenis_kelamin;
    };

    // Setter and Getter
    public void setNik(String nik) {
        this.nik = nik;
    }
    public String getNik() {
        return this.nik;
    }
    public void setNama(String nama) {
        this.nama = nama;
    }
    public String getNama() {
        return this.nama;
    }
    public void setJenisKelamin(String jenis_kelamin) {
        this.jenis_kelamin = jenis_kelamin;
    }
    public String getJenisKelamin() {
        return this.jenis_kelamin;
    }
    public void eat() {
        System.out.println(this.nama + " is Eating");
    }

    public void sleep() {
        System.out.println(this.nama + " is sleeping... (zzz)");
    }

}

File: CPP/program/Human.cpp
#include "Human.hpp"

// default constructor
Human::Human()
{
    this->nik = "";
    this->nama = "";
    this->jenisKelamin = "";
}

// Parameterized constructor
Human::Human(string nik, string nama, string jenisKelamin)
{
    this->nik = nik;
    this->nama = nama;
    this->jenisKelamin = jenisKelamin;
}

// Setter and Getter
void Human::setNik(string nik)
{
    this->nik = nik;
}

string Human::getNik()
{
    return this->nik;
}

void Human::setNama(string nama)
{
    this->nama = nama;
}

string Human::getNama()
{
    return this->nama;
}

void Human::setJenisKelamin(string getJenisKelamin)
{
    this->jenisKelamin = getJenisKelamin;
}

string Human::getJenisKelamin()
{
    return this->jenisKelamin;
}

// Just in case testing the class attribute
void Human::introduce()
{
    cout << "Halo saya " + this->nama + " saya seorang " + this->jenisKelamin + ", saya mempunyai nomor induk kependukukan " + this->nik;
}

// Destrurtor, not used
Human::~Human()
{
}